rules/Php85/Rector/FuncCall/ArrayKeyExistsNullToEmptyStringRector.php

```diff
@@ -0,0 +1,252 @@
+
+/**
+ * @see https://wiki.php.net/rfc/deprecations_php_8_5#deprecate_using_values_null_as_an_array_offset_and_when_calling_array_key_exists
+ * @see \Rector\Tests\Php85\Rector\FuncCall\RemoveFinfoBufferContextArgRector\RemoveFinfoBufferContextArgRectorTest
+ */
+final class ArrayKeyExistsNullToEmptyStringRector extends AbstractRector implements MinPhpVersionInterface
+{
+    public function __construct(
+        private readonly ReflectionResolver $reflectionResolver,
+        private readonly PropertyFetchAnalyzer $propertyFetchAnalyzer,
+        private readonly ValueResolver $valueResolver
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replace null key in array_key_exists with empty string',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+        array_key_exists($key, $array);
+        CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+        array_key_exists((string)$key, $array);
+        CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [FuncCall::class];
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        if(!$node instanceof FuncCall){
+            return null;
+        }
+        
+        if ($node->isFirstClassCallable()) {
+            return null;
+        }
+
+        if (! $this->isName($node, 'array_key_exists')) {
+            return null;
+        }
+
+        $scope = $node->getAttribute(AttributeKey::SCOPE);
+        if (! $scope instanceof Scope) {
+            return null;
+        }
+
+        $args = $node->getArgs();
+
+        $classReflection = $scope->getClassReflection();
+        $isTrait = $classReflection instanceof ClassReflection && $classReflection->isTrait();
+
+        $functionReflection = $this->reflectionResolver->resolveFunctionLikeReflectionFromCall($node);
+        if (! $functionReflection instanceof FunctionReflection) {
+            return null;
+        }
+
+        $parametersAcceptor = ParametersAcceptorSelectorVariantsWrapper::select($functionReflection, $node, $scope);
+        $isChanged = false;
+
+        $result = $this->processNullToStrictStringOnNodePosition(
+            $node,
+            $args,
+            0,
+            $isTrait,
+            $scope,
+            $parametersAcceptor
+        );
+        if ($result instanceof Node) {
+            $node = $result;
+            $isChanged = true;
+        }
+
+        if ($isChanged) {
+            return $node;
+        }
+
+        return null;
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersionFeature::DEPRECATE_NULL_ARG_IN_ARRAY_KEY_EXISTS_FUNCTION;
+    }
+
+    /**
+     * @param Arg[] $args
+     * @param int|string $position
+     */
+    private function processNullToStrictStringOnNodePosition(
```

Comment:
this method looks duplicated with `NullToStrictStringFuncCallArgRector`, can be moved to shared service imo

Reply:
I think I should create a class like this, right?
```php
final class NullToStrictStringConverter
{
    public function convertIfNull(Node $expr): Node
    {}
}
```
Where should I put it?”

Reply:
Yes, on Php81 namespace since it start there 

https://github.com/rectorphp/rector-src/tree/main/rules/Php81/NodeManipulator